datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -202,30 +205,39 @@ impl<F: FileOpener> FileStream<F> {
         })
     }
 
+    fn next_file(&mut self) -> Option<Result<(FileOpenFuture, Vec<ScalarValue>)>> {
```

Comment:
```suggestion
    // Begin opening the next file in parallel while decoding the current file in FileStream. 
    // Since file opening is mostly IO (and may involve a 
    // bunch of sequential IO), it can be parallelized with decoding. 
    fn next_file(&mut self) -> Option<Result<(FileOpenFuture, Vec<ScalarValue>)>> {
```

Also, I wonder what you think about calling this function `start_next_file` or something to hint that it begins to open the file ?